src/lib/ui/ClimateSettingForm/ClimateSettingScheduleForm.tsx

```diff
@@ -80,13 +91,38 @@ function Content({
     | 'climate_setting'
   >('device_select')
 
+  const returnToDeviceSelect = (): void => {
+    resetField('deviceId')
+    setPage('device_select')
+  }
+
+  useEffect(() => {
+    if (page === 'device_select' && device?.properties !== undefined) {
+      const defaultSetting = (device?.properties as ThermostatDeviceProperties)
+        .default_climate_setting
+      if (defaultSetting !== undefined) setPage('name_and_schedule')
+      else setPage('default_setting')
+    }
+  }, [device?.properties, page, setPage])
+
```

Comment:
From Chat over Huddle: It's probably better to do this in an onClick handler rather than using an effect